packages/imperative/src/utilities/src/CliUtils.ts

```diff
@@ -378,20 +392,14 @@ export class CliUtils {
      * @memberof CliUtils
      */
     public static showMsgWhenDeprecated(handlerParms: IHandlerParameters) {
-        if (handlerParms.definition.deprecatedReplacement) {
+        if (handlerParms.definition.deprecatedReplacement || handlerParms.definition.deprecatedReplacement === "") {
             // form the command that is deprecated
-            let oldCmd: string | number;
-            if (handlerParms.positionals.length >= 1) {
-                oldCmd = handlerParms.positionals[0];
-            }
-            if (handlerParms.positionals.length >= 2) {
-                oldCmd = oldCmd + " " + handlerParms.positionals[1];
-            }
-
+            const oldCmd = handlerParms.positionals.join(" ");
             // display the message
             handlerParms.response.console.error("\nWarning: The command '" + oldCmd + "' is deprecated.");
-            handlerParms.response.console.error("Recommended replacement: " +
-                handlerParms.definition.deprecatedReplacement);
+            // Use consolidated deprecated message logic
+            const deprecatedMessage = CliUtils.generateDeprecatedMessage(handlerParms.definition);
+            handlerParms.response.console.error(deprecatedMessage);
```

Comment:
Previously the `deprecatedReplacement` was printed as-is, now with the consolidated logic it will have newlines removed: `cmdDefinition.deprecatedReplacement.replace(/\n/g, " ")`. Is this change intentional?

Reply:
Previously `DefaultHelpGenerator` stripped out new lines when parsing the deprecation message, and `CommandProcessor` kept them. I like that the new behavior is consistent 👍 

https://github.com/zowe/zowe-cli/blob/e10af2e2eaeafcbe3f68762405746a8b801bc7b0/packages/imperative/src/utilities/src/CliUtils.ts#L376

Not sure which we prefer - stripping out new lines vs keeping them. It may not matter much since I don't recall seeing a deprecation message that contains new lines 😋